classes/class.usrtoHelper.php

```diff
@@ -157,12 +158,37 @@ protected function checkAccess($usr_id, $take_over_id, $group_id)
             return true;
         }
 
-        // If the user taking over is of id 13? or is not in the admin role he does not have permission.
-        if (!isset($usr_id) || $usr_id == 13 || !in_array(2, self::dic()->rbacreview()->assignedGlobalRoles($usr_id))) {
+        if(!$this->checkPluginAccess($usr_id)){
             ilUtil::sendFailure(self::plugin()->translate('no_permission'), true);
             ilUtil::redirect('login.php');
+        }
+    }
 
+    public  function checkPluginAccess($usr_id=null){
+        if (!isset($usr_id)) {
+            $usr_id = self::dic()->user()->getId();
```

Comment:
And here is the side effect, now this method depends on the state of self::dic()->user(), pass this from the outside always and get rid if this if here.

Reply:
done